pytorch_lightning/trainer/supporters.py

```diff
@@ -120,26 +120,134 @@ def mean(self):
 
 
 class PredictionCollection(object):
-    def __init__(self, global_rank: int, world_size: int):
+
+    ID_KEY = 'id'
+
+    def __init__(self, global_rank: int, world_size: int, all_gather_fn: Callable):
         self.global_rank = global_rank
         self.world_size = world_size
-        self.predictions = {}
-        self.num_predictions = 0
+        self.all_gather_fn = all_gather_fn
+        self._legacy_predictions = {}
+        self._predictions = {stage: {} for stage in LoggerStages}
```

Comment:
shall we use this instead?
https://github.com/PyTorchLightning/pytorch-lightning/blob/61308138c33ab2d52c84e72eab3a8f1c83fa23fd/pytorch_lightning/trainer/states.py#L39